packages/subscriptions/__tests__/integration.jest.js

```diff
@@ -78,5 +81,165 @@ describe('subscriptions', () => {
+    ).toBe(subscription.id)
+
+    await unsubscribe(subscription.id, context)
+    expect(await pgdb.public.subscriptions.count()).toBe(1)
+    expect(await pgdb.public.subscriptions.findOne({
+      id: subscription.id,
+      active: false
+    })).toBeTruthy()
+
+    expect(await getSubscriptionsForUser(
+      subProps.userId,
+      subscriberContext
+    ).then( a => a.length)).toBe(0)
+
+    expect(await getSubscriptionsForUser(
+      subProps.userId,
+      subscriberContext,
+      { includeNotActive: true }
+    ).then( a => a.length)).toBe(1)
+  }),
+
+  test('comments with filters', async () => {
+    const {
+      upsertSubscription,
+      getUnreadNotificationsForUserAndObject
+    } = require('../lib/Subscriptions')
+    const createDiscussion = require('@orbiting/backend-modules-discussions/graphql/resolvers/_mutations/createDiscussion')
+    const submitComment = require('@orbiting/backend-modules-discussions/graphql/resolvers/_mutations/submitComment')
+
+    const { context } = global.instance
+    const { pgdb } = context
+    const [editor] = createUsers(2, ['editor'])
+    const [author, subscriber] = createUsers(2, ['member', 'debater'])
+
+    const filters = [null, 'Comment', 'Document']
+
+    const discussionId = await createDiscussion(
+      null,
+      {
+        title: 'test',
+        anonymity: 'ALLOWED'
+      },
+      { ...context, user: editor }
+    )
+
+    for(let filter of filters) {
+      await Promise.all([
+        pgdb.public.subscriptions.truncate({ cascade: true }),
+        pgdb.public.events.truncate({ cascade: true }),
+        pgdb.public.subscriptions.truncate({ cascade: true }),
+        pgdb.public.notifications.truncate({ cascade: true })
+      ])
+
+      const subscription = await upsertSubscription(
+        {
+          userId: subscriber.id,
+          objectId: author.id,
+          type: 'User',
+          ...filter ? { filters: [filter] } : {}
+        },
+        { ...context, user: subscriber }
+      )
+      expect(subscription).toBeTruthy()
+
+      const comment = await submitComment(
+        null,
+        {
+          discussionId,
+          content: `test filter:${filter}`
+        },
+        { ...context, user: author }
+      )
+
+      const events = await pgdb.public.events.find({
+        objectType: 'Comment',
+        objectId: comment.id
+      })
+      expect(events.length).toEqual(filter === 'Document' ? 0 : 1)
+
+      const notifications = await pgdb.public.notifications.find({
+        userId: subscriber.id,
+        subscriptionId: subscription.id
+      })
+      expect(notifications.length).toEqual(filter === 'Document' ? 0 : 1)
+      expect(notifications[0]?.readAt).toBeFalsy()
+
+      expect(await getUnreadNotificationsForUserAndObject(
+        subscriber.id,
+        {
+          type: 'Comment',
+          id: comment.id
+        },
+        context
+      ).then( a => a?.length)).toBe(filter === 'Document' ? 0 : 1)
+    }
```

Comment:
👏 

Thinking of it, these two test would probably help, too:
- Test to see if author in document works (notification)
- Test to see which trumps author in document over format (notification)

Reply:
(Even unit tests instead of integration, assuming it is quite the feat to prepare.)

Reply:
checkout: servers/publikator/__tests__/document.subscriptions.int.jest.js